### PR TITLE
Lra 908 room ready rover interface form for common attributes

### PR DESCRIPTION
**Pull request by Mohammad4844**
Changes:
- Added Rover Form for Common Attribute States
- Added validations
  - if there is a checkbox option, one option needs to be selected
  - if there is a quantity box, a number needs to be added
- Added checks for the page (tentative). Cannot be accessed if:
  - room doesnt exist
  - room state doesnt exist for room for today (this is assumed/prerequisite since the step before this is saving RoomState)
  - if Common Attribute States have already been saved for today
- Interface looks nice on mobile
- main link is `/common_attribute_states/new?room_id=<ROOM_ID>`

To test:
1. Pick a room, go to its #show page
2. Click on the 'Edit Common Attribute States' link (note that this link is temporary)
3. This will throw an error: room_state needs to be created
4. In the rails console, run: `Room.find(<ROOM_ID>).room_states.create` (to create a RoomState for that specific room)
5. Click on the link again. It should take you to the form
6. Fill the whole form (validations are present, so everything needs to be filled/selected)
7. Save
8. Takes you back to the room, and common_attributes are created (changes are reflected in the database)
9. Clicking on the link again will throw an error, since you have already saved Common Attribute States for today

### CODE REVIEW

File: app/controllers/common_attribute_states_controller.rb [hunk 1]
```diff
@@ -1,71 +1,61 @@
...
-  # GET /common_attribute_states/1 or /common_attribute_states/1.json
-  def show
-  end
+  before_action :set_room, only: %i[ new create ]
 
   # GET /common_attribute_states/new
   def new
-    @common_attribute_state = CommonAttributeState.new
-  end
+    authorize CommonAttributeState.new
```
Comment (britaumich) [inline app/controllers/common_attribute_states_controller.rb:7]: `authorize CommonAttributeState is enough`, there is no need to add .new

File: app/controllers/common_attribute_states_controller.rb [hunk 2]
```diff
@@ -1,71 +1,61 @@
...
-  def edit
+    @common_attribute_states = CommonAttribute.all.map do |common_attribute|
+      common_attribute.common_attribute_states.new
+    end
   end
 
   # POST /common_attribute_states or /common_attribute_states.json
   def create
-    @common_attribute_state = CommonAttributeState.new(common_attribute_state_params)
+    authorize CommonAttributeState.new
```
Comment (britaumich) [inline app/controllers/common_attribute_states_controller.rb:16]: the same: `authorize CommonAttributeState is enough`, there is no need to add .new

**Review by palinaskakun — APPROVED** [review]
pulled and tested, looks good, thank you for writing detailed instructions

File: app/policies/common_attribute_state_policy.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  def update?
+    user_in_admin_group?
+  end
+  
+  def edit?
+    update?
+  end
+
+  def destroy?
```
Comment (britaumich) [inline app/policies/common_attribute_state_policy.rb:26]: Please remove methods that are not used for authorization. Leave create? and new? 
If we add more methods to the controller we'll add them to the policy file too.

File: app/views/common_attribute_states/new.html.erb [hunk 4]
```diff
@@ -1,9 +1,7 @@
-<h1>New common attribute state</h1>
+<h1>Common Attribute States</h1>
 
-<%= render "form", common_attribute_state: @common_attribute_state %>
+<h2><%= @room.floor.building.zone ? "Zone: #{@room.floor.building.zone.name}<br>" : ''  %></h2>
```
Comment (britaumich) [inline app/views/common_attribute_states/new.html.erb:3]: Please remove `<br>` from the line 3

**Review by britaumich — APPROVED** [review]
Good job on the ticket.
When other parts of the rover workflow are created, we'll return to these pages to link them to the rest of the forms.

**Pull request merged**